app/services/discovery_engine/sync/locking.rb

```diff
@@ -49,15 +49,15 @@ def with_locked_document(content_id, payload_version:, &critical_section)
       critical_section.call
     end
 
-    def payload_newer_than_remote?(content_id, payload_version:)
+    def outdated_payload_version?(content_id, payload_version:)
       # Sense check: This shouldn't ever come through as nil from Publishing API, but if it does,
       # the only really useful thing we can do is ignore this check entirely because we can't
       # meaningfully make a comparison.
-      return true if payload_version.nil?
+      return false if payload_version.nil?
 
       # If there is no remote version yet, our version is always newer by definition
       remote_version = latest_synced_version(content_id)
-      return true if remote_version.nil?
+      return false if remote_version.nil?
 
       remote_version.to_i >= payload_version.to_i
```

Comment:
⬆️ This bit is _already_ correct for the new method name and behaviour (but wasn't previously!)

Reply:
Oops! I missed that too. The new name is much better too.

Reply:
I think that might have been caught by unit tests for the `payload_newer_than_remote?` method.

I think some should be added here so that we're sure something else hasn't been missed.

Reply:
Yeah – that method is really quite functional compared to the other ones and tests should be quite easy.

Having slept on it there's probably a better design in general for this module, will add some tests for this one in particular and then maybe revisit some of the other ones in the afternoon!